src/unix/udp.c

```diff
@@ -660,6 +754,34 @@ int uv_udp_set_membership(uv_udp_t* handle,
   }
 }
 
+
+int uv_udp_set_source_membership(uv_udp_t* handle,
+                                 const char* multicast_addr,
+                                 const char* interface_addr,
+                                 const char* source_addr,
+                                 uv_membership membership) {
+  struct sockaddr_in mcast_addr4;
+  struct sockaddr_in src_addr4;
+  struct sockaddr_in6 mcast_addr6;
+  struct sockaddr_in6 src_addr6;
```

Comment:
I think you could use 2 `sockaddr_storage` that you could later on cast to `sockaddr_in` pr `sockaddr_in6`. See: `uv_udp_set_multicast_interface` implementation.

Reply:
I don't think so because the `uv_udp_set_source_membership` function should reproduce `uv_udp_set_membership` functionality with additional parameter `source_addr`. These functions have the same behavior and they should be written in the same manner IMHO.

Reply:
Ok, I'll take a look!